### ISSUE TITLE
tensorboard evaluator/RawEpisodeReturn question

### ISSUE THREAD

**Issue by mnguyen0226** [instadeepai/Mava#292]
Hi, I just have questions related to evaluator/RawEpisodeReturn. I have read through Mava paper (very well-written) and currently working on SMAC algorithms

1/ In the examples Google Colab, you mention that evaluator/RawEpisodeReturn 30-40 is are good score. Why is this the case? Also, is the max RawEpisodeReturn = 100 or can I change this number?

2/ For SMAC, in the paper, you mentioned that the algorithms VDN and MADQN, QMIX did not performed as well as PyMARL in the SMAC paper. I give it a shot and saw that the RawEpisodeReturn ~8-10 (but very unstable). Is the max evaluator/RawEpisodeReturn = 100? Else, what is the "good score" for SMAC in your environment during training?

Thanks!

Best,

**Comment by KaleabTessera** [comment]
Hi @mnguyen0226 , thanks for the questions. I am happy to elaborate. 

1. 30-40 was experimentally seen to be a good score, i.e. at that time most agents actively move towards their assigned targets. I don't believe there is an officialial max `RawEpisodeReturn` for this env, the reward is based on agent distance to goals (a simplified version of https://github.com/openai/multiagent-particle-envs/blob/master/multiagent/scenarios/simple_spread.py). @DriesSmit Do you know of a reasonable max to use for this environment?  
2. A good score on smac 3m is ~ 20 `RawEpisodeReturn`. We achieved this with vdn, with the following config:

config | setting
-- | --
architecture | DecentralisedValueActor
batch_size | 512
checkpoint | True
discount | 0.99
epsilon_decay | 0.00001
epsilon_min | 0.05
executor_fn | VDNFeedForwardExecutor
executor_variable_update_period | 1000
max_gradient_norm | 10
optim_decay | 0.99
optim_epsilon | 0.00001
optimizer | SGD
optimizer_lr | 0.01
shared_weights | True
target_update_period | 100
trainer_fn | VDNTrainer
network_type | feedforward
policy_networks_layer_sizes | [64, 64]
shared_weights | True

You might need to try a few random seeds to get faster convergence to this score. For smac, the max reward is calculated [here](https://github.com/oxwhirl/smac/blob/503a678c7378a08d5c55233cfa7ef967acfc93c5/smac/env/starcraft2/starcraft2.py#L256).

In the next few days, we will add winrate and other metrics so that it is easier to compare to other smac implementations. 

Does this answer all of your questions?

**Comment by mnguyen0226** [comment]
@KaleabTessera. Thank you for your response. I have trained the MADQN algoritms for 2 days now and it seems like the agent does not learn in all 23 environments. I will try VDN and will let you know if it improve the result

**Comment by KaleabTessera** [comment]
@mnguyen0226 We are improving our MADQN agent, we have a PR currently out for that.  :rocket: :raised_hands: 

Those are params for optims like RMSProp, e.g. 
```
snt.optimizers.RMSProp(
                learning_rate=lr, epsilon=optim_epsilon, decay=optim_decay
)
```

**Comment by mnguyen0226** [comment]
I see I see. I have not tested RMSProp yet but I set up as you recommended above with all VDN, QMIX, and DQN but got no luck in reproduce the result (got around 4-5 evaluation/rawEpisodeReturn and every fluctuate). I will test our other hyper-parameter and see if that improve. I acknowledge that it quites difficult to reproduce a paper like SMAC especially when they use different frameworks such as PyMARL.

Thanks!
